ivadomed/scripts/visualize_and_compare_testing_models.py

```diff
@@ -21,21 +21,7 @@
 
 matplotlib.rcParams['toolbar'] = 'None'  # Remove buttons
 
-gui_env = ['TKAgg', 'GTKAgg', 'Qt4Agg', 'WXAgg']
-selected_gui_env = []
-for gui in gui_env:
-    try:
-        matplotlib.use(gui)
-        from matplotlib import pyplot as plt
-
-        selected_gui_env = gui
-        break
-    except:
-        continue
-# If none works
-if selected_gui_env == []:
-    from matplotlib import pyplot as plt
-
+if matplotlib.get_backend() == "agg":
```

Comment:
I'm pretty sure this loop is redundant: https://matplotlib.org/stable/users/explain/backends.html

> Without a backend explicitly set, Matplotlib automatically detects a usable backend based on what is available on your system and on whether a GUI event loop is already running. The first usable backend in the following list is selected: MacOSX, QtAgg, GTK4Agg, Gtk3Agg, TkAgg, WxAgg, Agg.

and calling `use()` in the middle of tests is messing the tests up by leaking state from one test to another, because `use()` is setting an internal global variable and hence causing some tests to use TkAgg or QtAgg or just agg whatever might be chosen depending on the order the tests happen to run in.